dali/python/nvidia/dali/plugin/jax/iterator.py

```diff
@@ -246,7 +246,15 @@ def data_iterator_decorator(func):
         def create_iterator(*args, **wrapper_kwargs):
             pipeline_def_fn = pipeline_def(func)
 
+            if 'num_threads' not in wrapper_kwargs:
+                wrapper_kwargs['num_threads'] = 4
```

Comment:
This choice seems quite arbitrary. I'd recommend getting the number of processors (or perhaps the number of processors per GPU?).

Reply:
I did that because that's what we do it a lot in our codebase and especially in the tutorials. Do we have some benchmarks of what it does to the performance?

Reply:
But you need to account for the number of CPUs that may need to be used for the JAX itself, the speed of the network, and things that may run in parallel.
I'm afraid that without an elaborate autotune we can't provide an accurate value here.

Reply:
There's a lot of middle ground between elaborate autotune and a fixed number.
Still, I'd at least like to have a big, fat comment saying that it's subject to change.

Reply:
Done, added docs, comments and tutorial section on `nunm_threads`